esp-hal/src/spi/master.rs

```diff
@@ -2861,12 +3070,28 @@ impl Info {
         Ok(words)
     }
 
+    #[cfg_attr(place_spi_driver_in_ram, ram)]
+    async fn transfer_in_place_async(&self, words: &mut [u8]) -> Result<(), Error> {
+        for chunk in words.chunks_mut(FIFO_SIZE) {
+            self.write_bytes_async(chunk).await?;
+            self.read_bytes_from_fifo(chunk)?;
+        }
+
+        Ok(())
+    }
+
     fn start_operation(&self) {
         let reg_block = self.register_block();
         self.update();
         reg_block.cmd().modify(|_, w| w.usr().set_bit());
     }
 
+    async fn start_operation_async(&self) {
```

Comment:
This method name sounds like it only starts the operation but it waits until the operation is done.

How about `execute_async`? `run_async`?